src/ucp/core/ucp_mm.c

```diff
@@ -419,21 +420,24 @@ static ucs_status_t ucp_memh_register(ucp_context_h context, ucp_mem_h memh,
     }
 
     ucs_for_each_bit(md_index, md_map) {
+        tl_md = &context->tl_mds[md_index];
+
+        if (context->config.ext.reg_nb_mem_types & UCS_BIT(mem_type)) {
```

Comment:
minor: can introduce some aux var and initialize it outside the loop, e.g.
`reg_nb_mtype = context->config.ext.reg_nb_mem_types & UCS_BIT(mem_type)`
and then use it here and on line 468

Reply:
we could initialize the map of memory domains to register on during ucp context initialization according to nonblocking capabilities

Reply:
but we need to distinguish the cases when `context->config.ext.reg_nb_mem_types` is not set and when ODP is not supported by MD